src/libcore/char.rs

```diff
@@ -84,10 +84,17 @@ pub fn from_u32(i: u32) -> Option<char> {
     if (i > MAX as u32) || (i >= 0xD800 && i <= 0xDFFF) {
         None
     } else {
-        Some(unsafe { transmute(i) })
+        Some(unsafe { from_u32_unchecked(i) })
     }
 }
 
+/// Converts a `u32` to an `char`, not checking whether it is a valid unicode
+/// codepoint.
+#[inline]
+pub unsafe fn from_u32_unchecked(i: u32) -> char {
```

Comment:
Can you add an explicit `#[unstable]` annotation to this function? It should have its own feature name as well.